psoc/grobot_workspace/grobot_module_controller.cydsn/dht11.c

```diff
@@ -0,0 +1,234 @@
+
+uint8_t dht_get_temperature(){
+    
+    if(decoded_temperature_data == null){
+        return -1;
+    }
+    else{
+        return decoded_temperature_data; 
+    }    
+}    
+
+//What Return_Decoded_Humidity does:
+//    -Returns the data in decoded_humidity_data
+// 
+//Parameters: None
+//Returns: uint8_t decoded_humidity_data
+
+uint8_t dht_get_humidity(){
+    
+    if(decoded_humidity_data == null){
+        return -1;
+    }else{
+        return decoded_humidity_data;
+    }    
+}   
+
+
+//What delay_funct does:
+//    - Enable Delay_Timer_ISR to detect Delay_Tiemr Terminal Count interrupt
+//    - Start the Delay_Timer component.
+//    - This function does not use the CPU delay API, CyDelay(), but uses Delay_Timer to generate delay. 
+//
+//Parameters: None
+//Return: None
+
+void delay_funct(void){
+    count = 1;
+    executed = 0;
+    // Enable Delay_timer_ISR
+    Delay_Timer_ISR_Enable();
+    //Write 0 to Control_Reg_Delay_Timer to starting Delay_Timer component 
+    Control_Reg_Delay_Timer_Write(0);
+}    
+
+
+//What decode dht data does:
+//    - The value stored in countvalues_datatimer[] array, is used to decode the data of DHT output
+//    - The time difference between countvalues_datatimer[i] and countvalues_datatimer[i-1] is calculated
+//    - If the countvalues_datatimer[i] is approximately 70 counts, it is data 0, and if countvalues_datatimer[i] is approximately 120, it is data 1. 
+//
+//Parameters: None
+//Returns: None
+
+void decode_DHT_data(void){
+    int i;
+    for(i=0; i<44; i++){
+        diff_count_values[i]=countvalues_datatimer[i-1]-countvalues_datatimer[i];
+    }
+    
+    int j;
+    for(j=0; j<8; j++){
+        if(diff_count_values[j+3]>100){
+            humidity_values[7-j]=1;
+        }else{
+            humidity_values[7-j]=0;
+        }
+        
+        if(diff_count_values[j+19]>100){
+            temperature_values[7-j]=1;
+        }else{
+            temperature_values[7-j]=0;
+        }
+    }
+    decoded_temperature_data = (uint8_t)((temperature_values[7]<<7) + (temperature_values[6]<<6) + (temperature_values[5]<<5) + (temperature_values[4]<<4) + (temperature_values[3] <<3) + (temperature_values[2] <<2) + (temperature_values[1]<<1) + temperature_values[0]);
+    decoded_humidity_data=(uint8_t)((humidity_values[7]<<7) + (humidity_values[6]<<6) + (humidity_values[5]<<5) + (humidity_values[4]<<4) + (humidity_values[3] <<3) + (humidity_values[2] <<2) + (humidity_values[1]<<1) + humidity_values[0]);
+    
+    //Now the data is decoded
+    decode_flag = 1;
+    //From here we will have the acctual values. We can call functions from
+    //here or send it out depending on what we decide to go with 
+}    
+
+//Name: Delay_Timer_ISR_Handler
+//Summary:
+//    Interrupt Service Routine. Check the Delay_Timer status and clear the interrupt
+//    change the Count and Executed status
+
+_CY_ISR(Delay_Timer_ISR_Handler){
+    
+    if(Delay_Timer_ReadPeriod() == DELAY_ACQUISITION){
```

Comment:
What exactly does each status mean? I think a better comment in the summary section (or maybe here) would be an explanation of *why* we do this instead of a literal translation of what the function is doing.